OloEditor/SandboxProject/Assets/Scripts/LuaScripts/LuaGameManager.lua

```
@@ -0,0 +1,122 @@
+-- LuaGameManager.lua — Lua port of GameManager.cs
+-- Global game state: pause (Esc), death detection, victory condition, time scale.
+-- Attach via LuaScriptComponent with ScriptFile = "Scripts/LuaScripts/LuaGameManager.lua"
+
+local GameManager = {}
+
+local paused = false
+local gameOver = false
+local victory = false
+
+-- Shared game state table for other scripts to check
+GameState = GameState or {}
```

Luacheck (1.2.0)

[warning] setting non-standard global variable 'GameState'

(W111)